### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by mirouhml**
### General requirements

- Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
- Make sure that you used correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

### Ruby requirements
- Follow our list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).

### Create models
-  Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
-  Set up associations between models.

### Use models to insert data
- Create at least one user by running the following code:
    ```ruby
       first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
       second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.')
    ```
- Create at least 4 posts written by one of the users you created by running the following code:
    ```ruby
       first_post = Post.create(user: first_user, title: 'Hello', text: 'This is my first post')
    ```
- Create at least 6 posts comments for one of the posts you created by running the following code:
    ```ruby
       Comment.create(post: first_post, user: second_user, text: 'Hi Tom!' )
    ```

### Create custom methods
The models should include some additional methods.
-  Users
    - A method that returns the 3 most recent posts for a given user.
-  Posts
    - A method that updates the posts counter for a user.
    - A method which returns the 5 most recent comments for a given post.
-  Comments
    - A method that updates the comments counter for a post.
-  Likes
    - A method that updates the likes counter for a post.

Make current user data available to all controllers.

-  Create a method called current_user in your ApplicationController.
-  A current_user method should return the first user from a database.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts
+  has_many :comments
+  has_many :likes
```
Comment (arslanbisharat) [inline app/models/user.rb:4]: - [x] The associations between user models are set up correctly but according to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) you should have foreign keys 'author id'  for each association.

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :user
```
Comment (arslanbisharat) [inline app/models/post.rb]: - [x] According to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) you should have post.rb belongs to the author and it should have classname 'User'.

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :post
+  belongs_to :user
```
Comment (arslanbisharat) [inline app/models/comment.rb]: - [x] According to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) you should have comment.rb belongs to the author and it should have classname 'User'.

File: app/models/like.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :post
+  belongs_to :user
```
Comment (arslanbisharat) [inline app/models/like.rb]: - [x] According to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) you should have like.rb belongs to the author and it should have classname 'User'.

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
Hi @mirouhml, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- The Pull Request has a proper title and description. ✔️
- You have added a descriptive Readme file. Good Job. ✔️
- There is no linter error is present. ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mirouhml** [comment]
Hello @arslanbisharat thanks for the review, yes I get your point, according to the diagram the foreign keys should be mentioned as author_id instead of user_id, and author instead of user, but the commands and the requirements of the previous projects that lead to this, specified that we use user and user_id, as per the command of rails that adds the references for the tables, this is why I used that and I also specified that in my commands in the PR description.

**Comment by arslanbisharat** [comment]
> Hello @arslanbisharat thanks for the review, yes I get your point, according to the diagram the foreign keys should be mentioned as author_id instead of user_id, and author instead of user, but the commands and the requirements of the previous projects that lead to this, specified that we use user and user_id, as per the command of rails that adds the references for the tables, this is why I used that and I also specified that in my commands in the PR description.

Thank you so much for reaching out @mirouhml. I do understand your concern but I'm following the guidelines from Microverse Support Team Lead. I'll discuss this with them. You can also reach out to them using the form above. 

Your patience is appreciated.

**Comment by mirouhml** [comment]
Thanks for the quick reply @arslanbisharat

**Review by ShoiraTa — APPROVED** [review]
## **STATUS: APPROVED  🟢 ** 
------
### Hi @mirouhml , 

**Great work on making the changes requested by a previous reviewer 👏🏻** 

Your project is complete! 

5SW

**Cheers and Happy coding! :clap: :clap: :clap:**

Feel free to leave any questions or comments in the PR thread by mentioning my Github username if something is not 100% clear.
***
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have  limited number of  reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mirouhml** [comment]
@ShoiraTa thanks for the approval.

**Pull request merged**